server/public/model/config.go

```diff
@@ -4718,15 +4735,42 @@ func (o *Config) Sanitize(pluginManifests []*Manifest) {
 	}
 
 	for i := range o.SqlSettings.DataSourceReplicas {
-		o.SqlSettings.DataSourceReplicas[i] = FakeSetting
+		if partiallySanitizeDataSources {
+			sanitized, err := SanitizeDataSource(*o.SqlSettings.DriverName, o.SqlSettings.DataSourceReplicas[i])
+			if err != nil {
+				mlog.Warn("Failed to sanitize SqlSettings.DataSourceReplicas", mlog.Err(err))
+				continue
+			}
+			o.SqlSettings.DataSourceReplicas[i] = sanitized
+		} else {
+			o.SqlSettings.DataSourceReplicas[i] = FakeSetting
+		}
 	}
 
 	for i := range o.SqlSettings.DataSourceSearchReplicas {
-		o.SqlSettings.DataSourceSearchReplicas[i] = FakeSetting
+		if partiallySanitizeDataSources {
+			sanitized, err := SanitizeDataSource(*o.SqlSettings.DriverName, o.SqlSettings.DataSourceSearchReplicas[i])
+			if err != nil {
+				mlog.Warn("Failed to sanitize SqlSettings.DataSourceSearchReplicas", mlog.Err(err))
+				continue
+			}
+			o.SqlSettings.DataSourceSearchReplicas[i] = sanitized
+		} else {
+			o.SqlSettings.DataSourceSearchReplicas[i] = FakeSetting
+		}
 	}
 
 	for i := range o.SqlSettings.ReplicaLagSettings {
-		o.SqlSettings.ReplicaLagSettings[i].DataSource = NewPointer(FakeSetting)
+		if partiallySanitizeDataSources {
+			sanitized, err := SanitizeDataSource(*o.SqlSettings.DriverName, *o.SqlSettings.ReplicaLagSettings[i].DataSource)
+			if err != nil {
+				mlog.Warn("Failed to sanitize SqlSettings.ReplicaLagSettings", mlog.Err(err))
+				continue
+			}
+			o.SqlSettings.ReplicaLagSettings[i].DataSource = NewPointer(sanitized)
+		} else {
+			o.SqlSettings.ReplicaLagSettings[i].DataSource = NewPointer(FakeSetting)
+		}
```

Comment:
There's a bit of a repeating pattern here. Is there any way we could refactor a bit through some local util function perhaps?